File: lenguajes_pg/curso-cpp/03-funciones/recursividad/factorial.cpp
#include <iostream>
using namespace std;

int factorial(int numero);
int factorialRecursiva(int numero);

int main() {
    int numero;

    cout << "Ingrese un número: "; cin >> numero;

    cout << numero << "! = " << factorial(numero) << endl;
    cout << numero << "! = " << factorialRecursiva(numero) << endl;

    return 0;
}

int factorial(int numero) {
    int resultado = 1;

    for(int i=1;i<=numero;i++) {
        resultado *= i;
    }

    return resultado;
}

int factorialRecursiva(int numero) {
    return (numero == 0) ? 1 : numero * factorialRecursiva(numero - 1);
}

File: lenguajes_pg/curso-java/PrimerProyecto/src/estructurasControlFlujo/bucles/determinados/Factorial.java
package estructurasControlFlujo.bucles.determinados;

import javax.swing.JOptionPane;

public class Factorial {

	public static void main(String[] args) {
		long numero = Long.parseLong(JOptionPane.showInputDialog("Ingrese un número"));
		
		System.out.println(String.format("%d! = %d", numero, factorial(numero)));
		// System.out.println(String.format("%d! = %d", numero, factorialRecursiva(numero)));
	}

	public static long factorial(long numero) {
		long resultado = 1L;
		
		for(int i = 1; i <= numero; i++) {
			resultado *= i;
		}
		
		return resultado;
	}
	
	public static long factorialRecursiva(long numero) {
		return (numero == 0) ? 1 : numero * factorialRecursiva(numero - 1);
	}
}
